script.js

```diff
@@ -1,12 +1,91 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+window.onload = setTimeout(setup, 500);
+
+setTimeout(rootSetup, 500);
+
+// function getAllShows() {
+//   return fetch("https://api.tvmaze.com/shows")
+//     .then((response) => response.json())
+//     .then((data) => {
+//         data.forEach(({name, id}) => {
+//         availableShows.push({name, id});
+//       });
+//     })
+//     .catch((error) => console.log(error));
+// }
+
+// pull all episodes from the selected show
+
+function createShowList() {
+  let showList = [];
+  availableShows.forEach((show) => showList.push(show.name));
+  let select = document.querySelector("#show-selector");
+  showList.sort().forEach((show) => {
+    let option = document.createElement("option");
+    option.innerText = show;
+    select.appendChild(option);
+  });
 }
+setTimeout(createShowList, 300);
```

Comment:
Right now, you're saying "300ms after the page loads, assume `availableShows` has been populated, and make the show list". This works, but has a couple of problems:
1. Imagine you're on a really slow internet connection and it takes more than 300ms to get data from tv maze - this call will still happen, but it will render with no shows.
2. Imagine you're on a really _fast_ internet connection, and it only takes 10ms to get the data from tv maze - we'll still wait 300ms before we show anything on the screen, making our website slower than it could be.

Instead, if you call `createShowList()` as part of the callback from `fetch` completing, it'll happen at exactly the right time. It's the same deal with `setup` and `rootSetup`.